src/js/controllers/mapController.ts

```diff
@@ -81,6 +146,74 @@ export class MapController {
       });
   }
 
+  getMoreLayers(): Promise<any> {
+    const { appSettings } = store.getState();
+    const { layerPanel } = appSettings;
+    const detailedLayers: any = [];
+    const remoteDataLayers: any = [];
+
+    const layers = Object.keys(layerPanel)
+      .filter(groupName => {
+        return groupName !== 'GROUP_BASEMAP' && groupName !== 'extraLayers';
+      })
+      .reduce((list, groupName, groupIndex) => {
+        const orderedGroups = layerPanel[groupName].layers.map((layer: any) => {
+          return { groupId: groupName, ...layer };
+        });
+        return list.concat(orderedGroups);
+      }, []);
+
+    layers.forEach((layer: RemoteDataLayer) => {
+      if (layer.type === 'remoteDataLayer') {
+        remoteDataLayers.push({
+          order: layer.order,
+          layerGroupId: layer.groupId,
+          dataLayer: layer
+        });
+      } else {
+        detailedLayers.push(layer);
+      }
+    });
+
+    const remoteDataLayerRequests = remoteDataLayers.map(
+      (item: RemoteDataLayer, j: any) => {
+        return fetch(
+          `https://production-api.globalforestwatch.org/v1/layer/${item.dataLayer.uuid}`
+        )
+          .then(response => response.json())
+          .then(json => json.data)
+          .then(layer =>
+            fetch(layer.attributes.layerConfig.metadata)
+              .then(response => response.json())
+              .then(metadata => {
+                const attributes = layer.attributes;
+                const itemGroup = item.group;
+                // Object.keys(remoteDataLayers[j].layer).forEach(layerProp => {
+                //   if (layerProp !== 'type' && layerProp !== 'uuid') {
+                //     if (layerProp === 'legendConfig') {
+                //       attributes[layerProp] = remoteDataLayers[j].layer[layerProp];
+                //     } else {
+                //       layer.attributes.layerConfig[layerProp] = remoteDataLayers[j].layer[layerProp];
+                //     }
+                //   }
+                // });
+                item.layer = layer.attributes.layerConfig;
+                item.group = itemGroup;
+                item.layer.metadata = {
+                  metadata,
+                  legendConfig: attributes.legendConfig
+                };
+                return item;
+              })
```

Comment:
Missing error handling here: ` .catch(error => console.error(error));` This will cause some layers that do not return anything from API to be `undefined` which is exactly what we want to make sure we render all those that are returned and not all or nothing Promise.all approach!